HapticGenerator.swift

```diff
@@ -121,5 +166,82 @@ public struct Haptic {
             (generator as? UIFeedbackGenerator)?.prepare()
         }
     }
+}
+
+// MARK: - Extension that contains methods to determine current device & supported device(s)
+extension Haptic {
+    
+    // MARK: - Enums -
+    // MARK: Internal
     
+    /// An enum that represents the type of device (iPhone)
+    ///
+    /// - iPhone6S: An iPhone 6S
+    /// - iPhone6SPlus: An iPhone 6S Plus
+    /// - iPhone7: An iPhone 7
+    /// - iPhone7Plus: An iPhone 7 Plus
+    /// - iPhone8: An iPhone 8
+    /// - iPhone8Plus: An iPhone 8 Plus
+    /// - iPhoneX: An iPhone X
+    /// - unsupported: An unsupported device (iPhone 6 or below)
+    internal enum Device: String {
+        case iPhone6S
+        case iPhone6SPlus
+        case iPhone7
+        case iPhone7Plus
+        case iPhone8
+        case iPhone8Plus
+        case iPhoneX
+        case unsupported
+    }
+    
+    // MARK: - Methods -
+    // MARK: Internal
+    
+    /// Checks which device (iPhone model) the user has
+    ///
+    /// - Returns: A `Device` enum value that represents the user's device
+    internal func getCurrentDevice() -> Device {
+        var sysinfo = utsname()
+        uname(&sysinfo)
+        guard let model = String(bytes: Data(bytes: &sysinfo.machine, count: Int(_SYS_NAMELEN)), encoding: .ascii)?.trimmingCharacters(in: .controlCharacters) else {
+            return .unsupported
+        }
+        switch model {
+        case "iPhone10,3", "iPhone10,6":
```

Comment:
Hmm, maybe it would be better to skip checks for iPhone 7 or later. So as long as you have an iPhone 7 or later, `hasNewerTapticEngine()` would return `true`. Then you don't need to update the library every September!